### PR TITLE
Add support for externally managed NLB instances

### PR DESCRIPTION
**Pull request by falzm**
This change introduces a mechanism allowing operators to instruct the
CCM to skip mutation operations on externally managed NLB instances.

### CODE REVIEW

File: integtest/run.bash [hunk 1]
```diff
@@ -33,65 +37,25 @@ deploy_cluster() {
...
-}
-
-trap cleanup EXIT
-deploy_cluster
-deploy_ingress_controller
-deploy_test_app
-test_node_labels
-test_ingress_nlb
-test_node_expunge
+echo "=== ALL TESTS PASSED ==="
```
Comment (illi-j) [inline integtest/run.bash:61]: no newline at end of file

File: integtest/test-nlb-external.bash [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  -e "s/%%EXOSCALE_ZONE%%/$EXOSCALE_ZONE/" \
+  -e "s/%%EXTERNAL_NLB_ID%%/$EXTERNAL_NLB_ID/" \
+  -e "s/%%NODEPOOL_ID%%/$NODEPOOL_ID/" \
+  "${INTEGTEST_DIR}/manifests/hello-no-ingress.yml.tpl" \
+  | kubectl $KUBECTL_OPTS delete -f -
+_until_success "test \
+  \$(exo nlb show -z \$EXOSCALE_ZONE --output-template '{{.Services|len}}' \$EXTERNAL_NLB_ID) \
+  -eq 0"
+
+echo "<<< PASS"
```
Comment (illi-j) [inline integtest/test-nlb-external.bash:55]: same

**Review by illi-j — APPROVED** [review]
LGTM! we have huge integtest now, nice work, thx @falzm !

File: integtest/run.bash (hunk 1 above)
Comment (falzm) [reply integtest/run.bash:61]: Dammit, I always get tricked by this one... 😑

Comment (falzm) [reply integtest/run.bash]: Fixed in 94bd90e.

File: integtest/test-nlb-external.bash (hunk 2 above)
Comment (falzm) [reply integtest/test-nlb-external.bash]: Fixed in 94bd90e.

**Review by jessicatoscani — APPROVED** [review]

**Pull request merged**